File: Backtracking/NM3/Python_NM3/NM3.py
import sys


n, m = map(int, sys.stdin.readline().split())
num_list = []


def permutation_with_repetition(cnt):
    if cnt == m:
        print(*num_list)
        return

    for i in range(1, n + 1):
        num_list.append(i)
        permutation_with_repetition(cnt + 1)
        num_list.pop()


permutation_with_repetition(0)

File: Backtracking/NM3/Go_NM3/NM3.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const (
	max = 7 + 1
)

var (
	br      = bufio.NewReader(os.Stdin)
	bw      = bufio.NewWriter(os.Stdout)
	n, m    int
	numList [max]int
)

func permutationWithRepetition(cnt int) {
	if cnt == m {
		for i := 0; i < m; i++ {
			fmt.Fprintf(bw, "%d ", numList[i])
		}
		fmt.Fprintln(bw)
		return
	}

	for i := 1; i <= n; i++ {
		numList[cnt] = i
		permutationWithRepetition(cnt + 1)
	}
}

func main() {
	defer bw.Flush()

	fmt.Fscanln(br, &n, &m)

	permutationWithRepetition(0)
}
